lib/omni_auth/strategies/abstract_saml.rb

```diff
@@ -0,0 +1,101 @@
+        logout_responses_signed: true, # Enable signature on Logout Response
+        want_assertions_signed: true, # Require the IdP to sign its SAML Assertions
+        want_assertions_encrypted: true # Invalidate SAML messages without an EncryptedAssertion
+      }
+
+      # Our auto-login mechanism passes a desired redirect path (signed with a
+      # secret to prevent tampering) to the request phase.
+      # If we forward this via SAML's standard "RelayState" parameter, we will
+      # get it back in the callback phase.
+      # TODO: This parameter is not yet used in the app
+      option :idp_sso_target_url_runtime_params, {
+        redirect_path: 'RelayState'
+      }
+
+      # This Lambda is responsible for terminating the session and will
+      # only be used for identity-provider-initiated logout requests
+      option :idp_slo_session_destroy, (proc { |env, _session|
+        if Rails.env.development?
+          # For development purposes, we want to assume a secure connection behind an TLS-terminating proxy.
+          # If this option is not set, our session cookie is not sent by Rack (due to the `secure` flag)
+          env['HTTP_X_FORWARDED_PROTO'] = 'https'
+        end
+
+        # We logout the active user if possible. Probably, the cookie is not included
+        # in the request (due to a cross-origin request) and the following line is a no-op.
+        env['warden'].logout
+
+        # We want to ensure that the new cookie (with an empty session) is set by the client.
+        # Doing so will definitely overwrite an existing session in the browser
+        # Our mechanism will only work through HTTPS and with the following two options
+        env['rack.session'].options[:secure] = true
+        env['rack.session'].options[:same_site] = :none
+      })
+
+      info do
+        {
+          email: @attributes['urn:oid:0.9.2342.19200300.100.1.3'],
+          name: @attributes['urn:oid:2.5.4.3'],
+          first_name: @attributes['urn:oid:2.5.4.42'],
+          last_name: @attributes['urn:oid:2.5.4.4'],
+          display_name: @attributes['urn:oid:2.16.840.1.113730.3.1.241']
+        }
+      end
+
+      uid { @name_id || @attributes['urn:oid:0.9.2342.19200300.100.1.1'] }
+
+      def with_settings
+        # Get persistent IDs to recognize returning users
+        options[:name_identifier_format] = 'urn:oasis:names:tc:SAML:2.0:nameid-format:persistent'
+
+        options[:sp_entity_id] ||= hostname
+        options[:single_logout_service_url] ||= slo_path
+        super
+      end
+
+      def hostname
+        full_host + script_name
+      end
+
+      def slo_path
+        # This path is defined and handled by the `omniauth-saml` gem
+        "#{full_host}#{script_name}#{request_path}/slo"
+      end
+
+      def self.restore_deprecated_options
+        # The `omniauth-saml` gem still uses some of these deprecated options and does not work correctly otherwise
```

Comment:
I created a [PR at omniauth-saml](https://github.com/omniauth/omniauth-saml/pull/206) to remove these lines